### PR TITLE
#1831 Fixes multiple error in case of data class assertion inside assertsoftly

### PR DESCRIPTION
**Pull request by ashishkujoy**

### CODE REVIEW

File: kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/DataClassEq.kt [hunk 1]
```diff
@@ -64,9 +65,12 @@ internal object DataClassEq : Eq<Any> {
...
             }
-         else
-            runCatching { actualPropertyValue shouldBe expectedPropertyValue }
-               .fold(onSuccess = { null }, onFailure = { t -> Pair(prop, StandardDifference(t)) })
+         else {
+            eq(actualPropertyValue, expectedPropertyValue)
+               .toOption()
+               .map { Pair(prop, StandardDifference(it)) }
+               .orNull()
+         }
```
Comment (sksamuel) [inline kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/DataClassEq.kt:73]: Can you explain what the change is doing ?

Comment (ashishkujoy) [reply kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/DataClassEq.kt:73]: Earlier we are using `actualPropertyValue shouldBe expectedPropertyValue`

shouldBe internally add the error to errorCollector due to which we are getting multiple error message when data class contains enum, now i replaced that with eq function call which check equality and return error using which we are creating the difference message for data class without adding those extra errors in errorCollector.

Comment (ashishkujoy) [reply kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/DataClassEq.kt:73]: Is it good to merge?

Comment (sksamuel) [reply kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/eq/DataClassEq.kt:73]: Ok I see.

**Review by sksamuel — APPROVED** [review]

**Pull request merged**